### ISSUE TITLE
Use Connection/Edge Less Maybe?

### ISSUE THREAD

**Issue by 0xcaff** [forte-music/schema#5]
GraphQL allows for resolving and shipping only the requested data. Most of the lists connections are being used for can be replaced with full lists because they're usually short (<1k elements) and only the id will be queried initially with rich data requested as needed.

**Comment by 0xcaff** [comment]
Google Play Music even limits the amount of items in a playlist to a maximum of 1k to allow for fast and rich queue management and viewing. All items (ids) need to be known for them to be added to a queue. GPM also loads all needed information for a playlist in a single request.

**Comment by 0xcaff** [comment]
# Queue Management in Google Play Music
Google Play Music (GPM) has the following queue management features:
* add single songs to the queue
* add playlists to the queue (paylists are limited to 1k elements)
* add albums to the queue
* add stuff after current song or at end of queue
* generated queues (radio) (the next item in the queue is generated and shown 
* lazily and doesn't support insertion)
* long queues (play all songs in artist) (can't add these lists to the queue but
  can start playing from one. The entire queue is populated immediately.)

This is implemented by eagerly fetching a list of ids and information.  This
allows for viewing and modifying the queue while it is playing. This comes at
the cost of not being able to add undetermined lists to the queue (all songs by
search query, shuffling forever, etc.).

# Queue Management in Spotify
Spotify has the following queue management features:
* play a list of songs (artist, album, playlist, radio)
* can't insert songs after the current song or at end of queue

Spotify lazily fetches everything but it comes at the cost of usability. The
queues aren't even real.

# Queue Management in Forte
I would like to find the best of both worlds (useful queues and playing really
long lists). It will make a maximum of one request for the initial render of a 
list of songs (playlist, album, song search results). Forte will do something
similar to google play music and use both async generated queues and remain
usable.

In order to implement this, pagination is needed.

# The Case for Pagination
If pagination was removed, it would make fetching the initial data expensive.
The query sent will ask for all elements in the list with all data needed to
render the view (title, play count, album name, artist name, etc.). If this list
is too long the UI will be crippled. If the ids were fetched and then the rich
data, two requests would need to be made (not a good thing to do when you have
tools like graphql to define rich queries) one for the ids and one for the rich
info for the first few items. GPM avoids these massive lists by arbitrary limits
(max library 50k songs, max playlist 1k songs). Forte's goal is to be a music
service for personal collections, even massive ones).

With pagination, the first page of songs could be requested as full data and the
rest could be requested as ids or lazily page by page.

Some examples of long lists which need to render in a paginated way are long
playlists and entire library lists.

TL;DR: Just read the last section.
